### PR TITLE
Doc move to sphinx

### LINKED ISSUE #82: Style the website
Website: https://matplotlib.org/cheatsheets/

The website is currently contains only plain HTML.
It would be better if we can make it look more like the official site: https://matplotlib.org/stable/index.html

### PR DESCRIPTION
**Pull request by jklymak**
This moves the website to sphinx instead of raw html...

Closes #82

NewCheatsheets

### CODE REVIEW

**Comment by jklymak** [comment]
OK, this seems to work.  It would be nice to also add circleCI so that we could see the page before it built, but...

**Comment by rougier** [comment]
Is it possible to have cheatsheets side by side instead of top of each other?

**Comment by jklymak** [comment]
Almost certainly, but maybe as a follow up?  We have things side-by-side on the main site so I am sure we can figure out how to do that.

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,9 @@ handout-*.png
 # ----------------------------------
 figures/*.pdf
 fonts/**/*.[ot]tf
+
+# html build
+docs/_build/*
+
+# OS specific
+.DS_STore
```
Comment (QuLogic) [inline .gitignore:24]: It's
```suggestion
.DS_Store
```

**Review by QuLogic — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (jklymak) [reply .gitignore]: ooops, fixed!

**Comment by jklymak** [comment]
Changed the GitHub action to also upload the doc build.  

This PR would need to be merged before we can see CircleCI (and @tacaswell (or maybe @QuLogic) would have to grant it permission on the circleCI page).

**Comment by jklymak** [comment]
Latest commit makes the page with two and three columns:
BetterLayout

**Comment by QuLogic** [comment]
I'm not sure if we need to enable anything for Circle; it only triggers from config on the default branch, not PRs.

**Comment by jklymak** [comment]
I'm a little confused about our merge policy on this project.  

I think the policy should be: changes to the actual cheatsheets @rougier must approve, but changes to the CI and webpage can be done more generally by the Matplotlib team?

**Comment by jklymak** [comment]
@jimustafa Any comments/review on this?  You have been doing the bulk of the maintenance here...  Thanks!

**Comment by rougier** [comment]
@jklymak For cheatsheets, I think all the Matpltolib team should be able to approve PR, especially when it's not a big change. For bigger changes, I would be happy to keep an eye of it but nothing mandatory.

**Comment by jklymak** [comment]
@rougier OK, that sounds reasonable; we will be sure to request a review if something is major and/or aesthetic!

File: docs/index.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
+.. title:: Matplotlib cheatsheets
+
+***********************************
+Matplotlib cheatsheets and handouts
+***********************************
+
+Cheatsheets:
```
Comment (jimustafa) [inline docs/index.rst:7]: remove colon

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+         :width: 270px
+         :align: center
+         :alt: image of second page of cheatsheets
+         
+
+`Cheatsheets [pdf] <./cheatsheets.pdf>`_
+
+
+
+Handouts:
```
Comment (jimustafa) [inline docs/index.rst:32]: remove colon

File: .gitignore [hunk 4]
```diff
@@ -16,3 +16,9 @@ handout-*.png
 # ----------------------------------
 figures/*.pdf
 fonts/**/*.[ot]tf
+
+# html build
+docs/_build/*
+
+# OS specific
+.DS_Store
```
Comment (jimustafa) [inline .gitignore:24]: Might want to add an EOF newline character. Will need to update the pre-commit config to cover more of code base.

**Review by jimustafa — COMMENTED** [review]
The MPLBACKEND specification is a good idea for a more reproducible build, nice!

File: docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import datetime
+
+# -- Project information -----------------------------------------------------
+
+html_title = 'Visualization with Python'
```
[the remark this answers is not in the document]
Comment (jklymak) [reply docs/conf.py:5]: So, the actual title ends up being "Matplotlib Cheatsheets - Visualization with Python" so I think this is OK.

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+project = "Matplotlib cheatsheets"
+copyright = (
+     f"2012 - {datetime.datetime.now().year} The Matplotlib development team"
```
[the remark this answers is not in the document]
Comment (jklymak) [reply docs/conf.py:8]: I guess the appearance is that it should look like the rest of the docs, so the webpage should stay this way.  The cheatsheets may of course have their own copyright.

**Comment by jklymak** [comment]
> The MPLBACKEND specification is a good idea for a more reproducible build, nice!

Well, this was just because my conda env didn't have pyqt5 downloaded, and I realized we probably didn't want the build script to require a GUI framework.

**Comment by jklymak** [comment]
> > I'm not sure if we need to enable anything for Circle; it only triggers from config on the default branch, not PRs.
> 
> I had a similar question. It seems that the circle config is basically the same as for [matplotlib/mpl-brochure-site](https://github.com/matplotlib/mpl-brochure-site), and there the documents appear to be built and hosted for PRs. Could one of the checks for PRs here be a review of the built docs hosted on circle?

Yep, that was the plan; there will need to be a circle CI follow up as I'm sure I didn't do it properly, and we should add the action to post the link.

**Comment by tacaswell** [comment]
I'm going to merge this and then enable circle (the web UI won't let you enable a repo until it can see a config yaml in the branch you want to run off of).

**Comment by tacaswell** [comment]
https://app.circleci.com/pipelines/github/matplotlib/cheatsheets/1/workflows/b2554c3c-a13a-4ff0-919d-f2a34121b03e <- first circle build.

**Comment by jklymak** [comment]
Yes, circle is definitely broken.  Not sure how to install _all_ the deps....

**Pull request merged**